src/test/java/com/serotonin/bacnet4j/transport/ServiceFutureImplTest.java

```diff
@@ -0,0 +1,63 @@
+package com.serotonin.bacnet4j.transport;
+
+import static com.serotonin.bacnet4j.TestUtils.await;
+import static org.junit.Assert.assertThrows;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+
+import com.serotonin.bacnet4j.exception.BACnetException;
+import com.serotonin.bacnet4j.util.sero.ThreadUtils;
+
+public class ServiceFutureImplTest {
+    @Test(timeout = 10_000)
+    public void allWaitersAreReleased() throws Exception {
+        var sut = new ServiceFutureImpl();
+
+        var futures = new CompletableFuture[10];
+        var waiters = new AtomicInteger(0);
+        for(int i=0;i<10;i++) {
+            futures[i] = CompletableFuture.runAsync(() -> {
+                try {
+                    waiters.incrementAndGet();
+                    sut.get();
+                } catch (BACnetException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        }
+
+        // Wait for the other threads to start waiting on sut.get()
+        await(() -> waiters.get() == 10);
```

Comment:
I changed the sleeps to awaits since static sleep times have variable effectiveness in different environments.